// @wix/site-location/dist/cjs/location.context.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.locationContext = void 0;
const context_1 = require("@wix/sdk-runtime/context");
const location_runtime_1 = require("./location.runtime");
exports.locationContext = {
    baseUrl: (0, context_1.contextualizeHostModule)(location_runtime_1.locationRuntime, 'baseUrl'),
    path: (0, context_1.contextualizeHostModule)(location_runtime_1.locationRuntime, 'path'),
    prefix: (0, context_1.contextualizeHostModule)(location_runtime_1.locationRuntime, 'prefix'),
    protocol: (0, context_1.contextualizeHostModule)(location_runtime_1.locationRuntime, 'protocol'),
    query: (0, context_1.contextualizeHostModule)(location_runtime_1.locationRuntime, 'query'),
    url: (0, context_1.contextualizeHostModule)(location_runtime_1.locationRuntime, 'url'),
    onChange: (0, context_1.contextualizeHostModule)(location_runtime_1.locationRuntime, 'onChange'),
    to: (0, context_1.contextualizeHostModule)(location_runtime_1.locationRuntime, 'to'),
};
