// @wix/auto_sdk_data_movement-jobs/build/internal/es/meta.mjs
// src/hub-v1-job-movement-jobs.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixHubV1DataMovementServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/wix-data-hub-fs2",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/wix-data-hub-fs2",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/wix-data-hub-fs2",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/wix-data-hub-fs2",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/data-mover",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/wix-data-hub-fs2",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/wix-data-hub-fs2",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_data_movement-jobs";
function submitJob(payload) {
  function __submitJob({ host }) {
    const metadata = {
      entityFqdn: "wix.hub.v1.job",
      method: "POST",
      methodFqn: "wix.hub.v1.DataMovementService.SubmitJob",
      packageName: PACKAGE_NAME,
      url: resolveWixHubV1DataMovementServiceUrl({
        protoPath: "/v1/jobs",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "job.startedAt" }, { path: "job.finishedAt" }]
        }
      ])
    };
    return metadata;
  }
  return __submitJob;
}
function createFileUploadUrl(payload) {
  function __createFileUploadUrl({ host }) {
    const metadata = {
      entityFqdn: "wix.hub.v1.job",
      method: "POST",
      methodFqn: "wix.hub.v1.DataMovementService.CreateFileUploadUrl",
      packageName: PACKAGE_NAME,
      url: resolveWixHubV1DataMovementServiceUrl({
        protoPath: "/v1/temporary-files/upload-url",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __createFileUploadUrl;
}
function getJob(payload) {
  function __getJob({ host }) {
    const metadata = {
      entityFqdn: "wix.hub.v1.job",
      method: "GET",
      methodFqn: "wix.hub.v1.DataMovementService.GetJob",
      packageName: PACKAGE_NAME,
      url: resolveWixHubV1DataMovementServiceUrl({
        protoPath: "/v1/jobs/{jobId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "job.startedAt" }, { path: "job.finishedAt" }]
        }
      ])
    };
    return metadata;
  }
  return __getJob;
}
function listJobs(payload) {
  function __listJobs({ host }) {
    const metadata = {
      entityFqdn: "wix.hub.v1.job",
      method: "POST",
      methodFqn: "wix.hub.v1.DataMovementService.ListJobs",
      packageName: PACKAGE_NAME,
      url: resolveWixHubV1DataMovementServiceUrl({
        protoPath: "/v1/jobs/list",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "jobs.startedAt" }, { path: "jobs.finishedAt" }]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixHubV1DataMovementServiceUrl({
            protoPath: "/v1/jobs/list",
            data: payload,
            host
          }),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __listJobs;
}
function queryJobs(payload) {
  function __queryJobs({ host }) {
    const metadata = {
      entityFqdn: "wix.hub.v1.job",
      method: "POST",
      methodFqn: "wix.hub.v1.DataMovementService.QueryJobs",
      packageName: PACKAGE_NAME,
      url: resolveWixHubV1DataMovementServiceUrl({
        protoPath: "/v1/jobs/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "jobs.startedAt" }, { path: "jobs.finishedAt" }]
        }
      ]),
      fallback: [
        {
          method: "POST",
          url: resolveWixHubV1DataMovementServiceUrl({
            protoPath: "/v1/jobs/query",
            data: payload,
            host
          }),
          data: payload
        }
      ]
    };
    return metadata;
  }
  return __queryJobs;
}
function terminateJob(payload) {
  function __terminateJob({ host }) {
    const metadata = {
      entityFqdn: "wix.hub.v1.job",
      method: "POST",
      methodFqn: "wix.hub.v1.DataMovementService.TerminateJob",
      packageName: PACKAGE_NAME,
      url: resolveWixHubV1DataMovementServiceUrl({
        protoPath: "/v1/jobs/{jobId}/terminate",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "job.startedAt" }, { path: "job.finishedAt" }]
        }
      ])
    };
    return metadata;
  }
  return __terminateJob;
}
function listMovementLogs(payload) {
  function __listMovementLogs({ host }) {
    const metadata = {
      entityFqdn: "wix.hub.v1.job",
      method: "GET",
      methodFqn: "wix.hub.v1.DataMovementService.ListMovementLogs",
      packageName: PACKAGE_NAME,
      url: resolveWixHubV1DataMovementServiceUrl({
        protoPath: "/v1/jobs/{jobId}/logs",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __listMovementLogs;
}

// src/hub-v1-job-movement-jobs.meta.ts
function submitJob2() {
  const payload = {};
  const getRequestOptions = submitJob(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/jobs",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createFileUploadUrl2() {
  const payload = {};
  const getRequestOptions = createFileUploadUrl(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/temporary-files/upload-url",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getJob2() {
  const payload = { jobId: ":jobId" };
  const getRequestOptions = getJob(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/jobs/{jobId}",
    pathParams: { jobId: "jobId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listJobs2() {
  const payload = {};
  const getRequestOptions = listJobs(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/jobs",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryJobs2() {
  const payload = {};
  const getRequestOptions = queryJobs(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/jobs/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function terminateJob2() {
  const payload = { jobId: ":jobId" };
  const getRequestOptions = terminateJob(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/jobs/{jobId}/terminate",
    pathParams: { jobId: "jobId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listMovementLogs2() {
  const payload = { jobId: ":jobId" };
  const getRequestOptions = listMovementLogs(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/jobs/{jobId}/logs",
    pathParams: { jobId: "jobId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  createFileUploadUrl2 as createFileUploadUrl,
  getJob2 as getJob,
  listJobs2 as listJobs,
  listMovementLogs2 as listMovementLogs,
  queryJobs2 as queryJobs,
  submitJob2 as submitJob,
  terminateJob2 as terminateJob
};
