// @wix/site-site/dist/esm/site.context.js
import { contextualizeHostModule } from "@wix/sdk-runtime/context";
import { siteRuntime as siteModule } from './site.runtime';
export const siteContext = {
    currency: contextualizeHostModule(siteModule, 'currency'),
    currentPage: contextualizeHostModule(siteModule, 'currentPage'),
    language: contextualizeHostModule(siteModule, 'language'),
    regionalSettings: contextualizeHostModule(siteModule, 'regionalSettings'),
    revision: contextualizeHostModule(siteModule, 'revision'),
    timezone: contextualizeHostModule(siteModule, 'timezone'),
    getFontsHtml: contextualizeHostModule(siteModule, 'getFontsHtml'),
    getSiteDisplayName: contextualizeHostModule(siteModule, 'getSiteDisplayName'),
    getSiteStructure: contextualizeHostModule(siteModule, 'getSiteStructure'),
    getSiteThemeHtml: contextualizeHostModule(siteModule, 'getSiteThemeHtml'),
    prefetchPageResources: contextualizeHostModule(siteModule, 'prefetchPageResources'),
    routerSitemap: contextualizeHostModule(siteModule, 'routerSitemap'),
};
