// @wix/wix-data-items-common/dist/cjs/api/index.js
"use strict";

exports.__esModule = true;
exports.WixDataUpdateOptions = exports.WixDataSearch = exports.WixDataResult = exports.WixDataReference = exports.WixDataReadWithProjectionOptions = exports.WixDataReadOptions = exports.WixDataQueryReferencedOptions = exports.WixDataQueryOptions = exports.WixDataQuery = exports.WixDataPatchOptions = exports.WixDataPatch = exports.WixDataOptions = exports.WixDataItem = exports.WixDataFilter = exports.WixDataBulkResult = exports.WixDataBulkPatch = exports.WixDataBulkError = exports.WixDataApi = exports.WixDataAggregateOptions = exports.WixDataAggregate = exports.QueryValidator = exports.QueryBase = void 0;
var _WixDataQuery = require("./WixDataQuery");
exports.WixDataQuery = _WixDataQuery.WixDataQuery;
var _WixDataSearch = require("./WixDataSearch");
exports.WixDataSearch = _WixDataSearch.WixDataSearch;
var _WixDataApi = require("./WixDataApi");
exports.WixDataApi = _WixDataApi.WixDataApi;
var _WixDataResult = require("./WixDataResult");
exports.WixDataResult = _WixDataResult.WixDataResult;
var _QueryValidator = require("./QueryValidator");
exports.QueryValidator = _QueryValidator.QueryValidator;
var _WixDataAggregate = require("./WixDataAggregate");
exports.WixDataAggregate = _WixDataAggregate.WixDataAggregate;
var _QueryBase = require("./QueryBase");
exports.QueryBase = _QueryBase.QueryBase;
var _WixDataPatch = require("./WixDataPatch");
exports.WixDataPatch = _WixDataPatch.WixDataPatch;
exports.WixDataBulkPatch = _WixDataPatch.WixDataBulkPatch;
var _WixDataFilter = require("./WixDataFilter");
exports.WixDataFilter = _WixDataFilter.WixDataFilter;
var _types = require("./types");
exports.WixDataOptions = _types.WixDataOptions;
exports.WixDataReadOptions = _types.WixDataReadOptions;
exports.WixDataReadWithProjectionOptions = _types.WixDataReadWithProjectionOptions;
exports.WixDataUpdateOptions = _types.WixDataUpdateOptions;
exports.WixDataPatchOptions = _types.WixDataPatchOptions;
exports.WixDataQueryOptions = _types.WixDataQueryOptions;
exports.WixDataItem = _types.WixDataItem;
exports.WixDataReference = _types.WixDataReference;
exports.WixDataBulkError = _types.WixDataBulkError;
exports.WixDataBulkResult = _types.WixDataBulkResult;
exports.WixDataQueryReferencedOptions = _types.WixDataQueryReferencedOptions;
exports.WixDataAggregateOptions = _types.WixDataAggregateOptions;
