// @wix/site-window/dist/cjs/consentPolicy.runtime.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.consentPolicyRuntime = void 0;
const host_modules_1 = require("@wix/sdk-runtime/host-modules");
exports.consentPolicyRuntime = (0, host_modules_1.createHostModule)({
    getCurrentConsentPolicy: ({ channel }) => function getCurrentConsentPolicy(...args) {
        return channel.invoke({
            namespace: 'window',
            method: 'consentPolicy.getCurrentConsentPolicy',
            args,
        });
    },
    onConsentPolicyChanged: ({ channel }) => function onConsentPolicyChanged(...args) {
        return channel.invoke({
            namespace: 'window',
            method: 'consentPolicy.onConsentPolicyChanged',
            args,
        });
    },
    resetConsentPolicy: ({ channel }) => function resetConsentPolicy(...args) {
        return channel.invoke({
            namespace: 'window',
            method: 'consentPolicy.resetConsentPolicy',
            args,
        });
    },
    setConsentPolicy: ({ channel }) => function setConsentPolicy(...args) {
        return channel.invoke({
            namespace: 'window',
            method: 'consentPolicy.setConsentPolicy',
            args,
        });
    },
});
