// @wix/site-window/dist/cjs/context.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.window = exports.warmupData = exports.rendering = exports.multilingual = exports.lightbox = exports.consentPolicy = void 0;
var consentPolicy_context_1 = require("./consentPolicy.context");
Object.defineProperty(exports, "consentPolicy", { enumerable: true, get: function () { return consentPolicy_context_1.consentPolicyContext; } });
var lightbox_context_1 = require("./lightbox.context");
Object.defineProperty(exports, "lightbox", { enumerable: true, get: function () { return lightbox_context_1.lightboxContext; } });
var multilingual_context_1 = require("./multilingual.context");
Object.defineProperty(exports, "multilingual", { enumerable: true, get: function () { return multilingual_context_1.multilingualContext; } });
var rendering_context_1 = require("./rendering.context");
Object.defineProperty(exports, "rendering", { enumerable: true, get: function () { return rendering_context_1.renderingContext; } });
var warmupData_context_1 = require("./warmupData.context");
Object.defineProperty(exports, "warmupData", { enumerable: true, get: function () { return warmupData_context_1.warmupDataContext; } });
var window_context_1 = require("./window.context");
Object.defineProperty(exports, "window", { enumerable: true, get: function () { return window_context_1.windowContext; } });
