// @wix/site-window/dist/cjs/window.context.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.windowContext = void 0;
const context_1 = require("@wix/sdk-runtime/context");
const window_runtime_1 = require("./window.runtime");
exports.windowContext = {
    browserLocale: (0, context_1.contextualizeHostModule)(window_runtime_1.windowRuntime, 'browserLocale'),
    formFactor: (0, context_1.contextualizeHostModule)(window_runtime_1.windowRuntime, 'formFactor'),
    referrer: (0, context_1.contextualizeHostModule)(window_runtime_1.windowRuntime, 'referrer'),
    viewMode: (0, context_1.contextualizeHostModule)(window_runtime_1.windowRuntime, 'viewMode'),
    copyToClipboard: (0, context_1.contextualizeHostModule)(window_runtime_1.windowRuntime, 'copyToClipboard'),
    getAppPageData: (0, context_1.contextualizeHostModule)(window_runtime_1.windowRuntime, 'getAppPageData'),
    getBoundingRect: (0, context_1.contextualizeHostModule)(window_runtime_1.windowRuntime, 'getBoundingRect'),
    getCurrentGeolocation: (0, context_1.contextualizeHostModule)(window_runtime_1.windowRuntime, 'getCurrentGeolocation'),
    getRouterData: (0, context_1.contextualizeHostModule)(window_runtime_1.windowRuntime, 'getRouterData'),
    openAppLightbox: (0, context_1.contextualizeHostModule)(window_runtime_1.windowRuntime, 'openAppLightbox'),
    openLightbox: (0, context_1.contextualizeHostModule)(window_runtime_1.windowRuntime, 'openLightbox'),
    openModal: (0, context_1.contextualizeHostModule)(window_runtime_1.windowRuntime, 'openModal'),
    postMessage: (0, context_1.contextualizeHostModule)(window_runtime_1.windowRuntime, 'postMessage'),
    registerEventListener: (0, context_1.contextualizeHostModule)(window_runtime_1.windowRuntime, 'registerEventListener'),
    scrollBy: (0, context_1.contextualizeHostModule)(window_runtime_1.windowRuntime, 'scrollBy'),
    scrollTo: (0, context_1.contextualizeHostModule)(window_runtime_1.windowRuntime, 'scrollTo'),
    trackEvent: (0, context_1.contextualizeHostModule)(window_runtime_1.windowRuntime, 'trackEvent'),
};
