// @wix/site-storage/dist/cjs/local.runtime.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.localRuntime = void 0;
const host_modules_1 = require("@wix/sdk-runtime/host-modules");
exports.localRuntime = (0, host_modules_1.createHostModule)({
    clear: ({ channel }) => function clear(...args) {
        return channel.invoke({
            namespace: 'storage',
            method: 'local.clear',
            args,
        });
    },
    getItem: ({ channel }) => function getItem(...args) {
        return channel.invoke({
            namespace: 'storage',
            method: 'local.getItem',
            args,
        });
    },
    removeItem: ({ channel }) => function removeItem(...args) {
        return channel.invoke({
            namespace: 'storage',
            method: 'local.removeItem',
            args,
        });
    },
    setItem: ({ channel }) => function setItem(...args) {
        return channel.invoke({
            namespace: 'storage',
            method: 'local.setItem',
            args,
        });
    },
});
