// @wix/site-storage/dist/cjs/session.runtime.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.sessionRuntime = void 0;
const host_modules_1 = require("@wix/sdk-runtime/host-modules");
exports.sessionRuntime = (0, host_modules_1.createHostModule)({
    clear: ({ channel }) => function clear(...args) {
        return channel.invoke({
            namespace: 'storage',
            method: 'session.clear',
            args,
        });
    },
    getItem: ({ channel }) => function getItem(...args) {
        return channel.invoke({
            namespace: 'storage',
            method: 'session.getItem',
            args,
        });
    },
    removeItem: ({ channel }) => function removeItem(...args) {
        return channel.invoke({
            namespace: 'storage',
            method: 'session.removeItem',
            args,
        });
    },
    setItem: ({ channel }) => function setItem(...args) {
        return channel.invoke({
            namespace: 'storage',
            method: 'session.setItem',
            args,
        });
    },
});
