// @wix/site-site/dist/cjs/site.context.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.siteContext = void 0;
const context_1 = require("@wix/sdk-runtime/context");
const site_runtime_1 = require("./site.runtime");
exports.siteContext = {
    currency: (0, context_1.contextualizeHostModule)(site_runtime_1.siteRuntime, 'currency'),
    currentPage: (0, context_1.contextualizeHostModule)(site_runtime_1.siteRuntime, 'currentPage'),
    language: (0, context_1.contextualizeHostModule)(site_runtime_1.siteRuntime, 'language'),
    regionalSettings: (0, context_1.contextualizeHostModule)(site_runtime_1.siteRuntime, 'regionalSettings'),
    revision: (0, context_1.contextualizeHostModule)(site_runtime_1.siteRuntime, 'revision'),
    timezone: (0, context_1.contextualizeHostModule)(site_runtime_1.siteRuntime, 'timezone'),
    getFontsHtml: (0, context_1.contextualizeHostModule)(site_runtime_1.siteRuntime, 'getFontsHtml'),
    getSiteDisplayName: (0, context_1.contextualizeHostModule)(site_runtime_1.siteRuntime, 'getSiteDisplayName'),
    getSiteStructure: (0, context_1.contextualizeHostModule)(site_runtime_1.siteRuntime, 'getSiteStructure'),
    getSiteThemeHtml: (0, context_1.contextualizeHostModule)(site_runtime_1.siteRuntime, 'getSiteThemeHtml'),
    prefetchPageResources: (0, context_1.contextualizeHostModule)(site_runtime_1.siteRuntime, 'prefetchPageResources'),
    routerSitemap: (0, context_1.contextualizeHostModule)(site_runtime_1.siteRuntime, 'routerSitemap'),
};
