// @wix/site-site/dist/cjs/wixRouterSitemapEntry.context.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.wixRouterSitemapEntryContext = void 0;
const context_1 = require("@wix/sdk-runtime/context");
const wixRouterSitemapEntry_runtime_1 = require("./wixRouterSitemapEntry.runtime");
exports.wixRouterSitemapEntryContext = {
    changeFrequency: (0, context_1.contextualizeHostModule)(wixRouterSitemapEntry_runtime_1.wixRouterSitemapEntryRuntime, 'changeFrequency'),
    lastModified: (0, context_1.contextualizeHostModule)(wixRouterSitemapEntry_runtime_1.wixRouterSitemapEntryRuntime, 'lastModified'),
    pageName: (0, context_1.contextualizeHostModule)(wixRouterSitemapEntry_runtime_1.wixRouterSitemapEntryRuntime, 'pageName'),
    priority: (0, context_1.contextualizeHostModule)(wixRouterSitemapEntry_runtime_1.wixRouterSitemapEntryRuntime, 'priority'),
    title: (0, context_1.contextualizeHostModule)(wixRouterSitemapEntry_runtime_1.wixRouterSitemapEntryRuntime, 'title'),
    url: (0, context_1.contextualizeHostModule)(wixRouterSitemapEntry_runtime_1.wixRouterSitemapEntryRuntime, 'url'),
};
