// @wix/site-site/dist/cjs/wixRouterSitemapEntry.runtime.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.wixRouterSitemapEntryRuntime = void 0;
const host_modules_1 = require("@wix/sdk-runtime/host-modules");
exports.wixRouterSitemapEntryRuntime = (0, host_modules_1.createHostModule)({
    changeFrequency: ({ channel }) => function changeFrequency(...args) {
        return channel.invoke({
            namespace: 'site',
            method: 'wixRouterSitemapEntry.changeFrequency',
            args,
        });
    },
    lastModified: ({ channel }) => function lastModified(...args) {
        return channel.invoke({
            namespace: 'site',
            method: 'wixRouterSitemapEntry.lastModified',
            args,
        });
    },
    pageName: ({ channel }) => function pageName(...args) {
        return channel.invoke({
            namespace: 'site',
            method: 'wixRouterSitemapEntry.pageName',
            args,
        });
    },
    priority: ({ channel }) => function priority(...args) {
        return channel.invoke({
            namespace: 'site',
            method: 'wixRouterSitemapEntry.priority',
            args,
        });
    },
    title: ({ channel }) => function title(...args) {
        return channel.invoke({
            namespace: 'site',
            method: 'wixRouterSitemapEntry.title',
            args,
        });
    },
    url: ({ channel }) => function url(...args) {
        return channel.invoke({
            namespace: 'site',
            method: 'wixRouterSitemapEntry.url',
            args,
        });
    },
});
