// @wix/auto_sdk_secrets_secrets/build/internal/es/meta.mjs
// src/velo-secrets-vault-v1-secret-secrets.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixVeloSecretsVaultV1SecretsVaultServiceUrl(opts) {
  const domainToMappings = {
    "service.wix-code.com": [
      {
        srcPath: "/_api/cloud-secrets-vault-server",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/cloud-secrets-vault-server",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/cloud-secrets-vault-server",
        destPath: ""
      }
    ],
    "service-int.wix-code.com": [
      {
        srcPath: "/_api/cloud-secrets-vault-server",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/_api/cloud-secrets-vault-server",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_secrets_secrets";
function getSecretValue(payload) {
  function __getSecretValue({ host }) {
    const metadata = {
      entityFqdn: "wix.velo.secrets_vault.v1.secret",
      method: "GET",
      methodFqn: "wix.velo.secrets_vault.v1.SecretsVaultService.GetSecretValue",
      packageName: PACKAGE_NAME,
      url: resolveWixVeloSecretsVaultV1SecretsVaultServiceUrl({
        protoPath: "/api/v1/secrets/name/{name}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getSecretValue;
}
function listSecretInfo(payload) {
  function __listSecretInfo({ host }) {
    const metadata = {
      entityFqdn: "wix.velo.secrets_vault.v1.secret",
      method: "GET",
      methodFqn: "wix.velo.secrets_vault.v1.SecretsVaultService.ListSecretInfo",
      packageName: PACKAGE_NAME,
      url: resolveWixVeloSecretsVaultV1SecretsVaultServiceUrl({
        protoPath: "/api/v1/secrets",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "secrets.createdDate" },
            { path: "secrets.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listSecretInfo;
}
function createSecret(payload) {
  function __createSecret({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "secret.createdDate" }, { path: "secret.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.velo.secrets_vault.v1.secret",
      method: "POST",
      methodFqn: "wix.velo.secrets_vault.v1.SecretsVaultService.CreateSecret",
      packageName: PACKAGE_NAME,
      url: resolveWixVeloSecretsVaultV1SecretsVaultServiceUrl({
        protoPath: "/api/v1/secrets",
        data: serializedData,
        host
      }),
      data: serializedData
    };
    return metadata;
  }
  return __createSecret;
}
function deleteSecret(payload) {
  function __deleteSecret({ host }) {
    const metadata = {
      entityFqdn: "wix.velo.secrets_vault.v1.secret",
      method: "DELETE",
      methodFqn: "wix.velo.secrets_vault.v1.SecretsVaultService.DeleteSecret",
      packageName: PACKAGE_NAME,
      url: resolveWixVeloSecretsVaultV1SecretsVaultServiceUrl({
        protoPath: "/api/v1/secrets/{id}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteSecret;
}
function partiallyUpdateSecret(payload) {
  function __partiallyUpdateSecret({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "secret.createdDate" }, { path: "secret.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.velo.secrets_vault.v1.secret",
      method: "PATCH",
      methodFqn: "wix.velo.secrets_vault.v1.SecretsVaultService.PartiallyUpdateSecret",
      packageName: PACKAGE_NAME,
      url: resolveWixVeloSecretsVaultV1SecretsVaultServiceUrl({
        protoPath: "/api/v1/secrets/{id}",
        data: serializedData,
        host
      }),
      data: serializedData
    };
    return metadata;
  }
  return __partiallyUpdateSecret;
}

// src/velo-secrets-vault-v1-secret-secrets.meta.ts
function getSecretValue2() {
  const payload = { name: ":name" };
  const getRequestOptions = getSecretValue(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/api/v1/secrets/name/{name}",
    pathParams: { name: "name" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listSecretInfo2() {
  const payload = {};
  const getRequestOptions = listSecretInfo(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/api/v1/secrets",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createSecret2() {
  const payload = {};
  const getRequestOptions = createSecret(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/api/v1/secrets",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteSecret2() {
  const payload = { id: ":id" };
  const getRequestOptions = deleteSecret(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/api/v1/secrets/{id}",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateSecret() {
  const payload = { id: ":id" };
  const getRequestOptions = partiallyUpdateSecret(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/api/v1/secrets/{id}",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  createSecret2 as createSecret,
  deleteSecret2 as deleteSecret,
  getSecretValue2 as getSecretValue,
  listSecretInfo2 as listSecretInfo,
  updateSecret
};
