// @wix/auto_sdk_data_tasks/build/internal/es/index.mjs
// src/data-autocms-v2-background-task-tasks.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/data-autocms-v2-background-task-tasks.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixCloudAutocmsTasksV1TaskServiceUrl(opts) {
  const domainToMappings = {
    "editor._base_domain_": [
      {
        srcPath: "/_api/autocms/v1/tasks",
        destPath: "/v1"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/autocms/v1/tasks",
        destPath: "/v1"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/autocms/v1/tasks",
        destPath: "/v1"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/autocms/v1/tasks",
        destPath: "/v1"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/autocms/v1/tasks",
        destPath: "/v1"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/autocms/v1/tasks",
        destPath: "/v1"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/cms/v1/tasks",
        destPath: "/v1"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/autocms/v1/tasks",
        destPath: "/v1"
      }
    ],
    _: [
      {
        srcPath: "/_api/autocms/v1/tasks",
        destPath: "/v1"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_data_tasks";
function createTask(payload) {
  function __createTask({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "task.startedAt" },
          { path: "task.finishedAt" },
          {
            path: "task.updatePublishStatusOptions.scheduleDraftStatusOptions.date"
          },
          {
            path: "task.updatePublishStatusOptions.schedulePublishedStatusOptions.date"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.data.autocms.v2.background_task",
      method: "POST",
      methodFqn: "wix.cloud.autocms.tasks.v1.TaskService.CreateTask",
      packageName: PACKAGE_NAME,
      url: resolveWixCloudAutocmsTasksV1TaskServiceUrl({
        protoPath: "/v1",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "task.startedAt" },
            { path: "task.finishedAt" },
            {
              path: "task.updatePublishStatusOptions.scheduleDraftStatusOptions.date"
            },
            {
              path: "task.updatePublishStatusOptions.schedulePublishedStatusOptions.date"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createTask;
}
function listTasks(payload) {
  function __listTasks({ host }) {
    const metadata = {
      entityFqdn: "wix.data.autocms.v2.background_task",
      method: "GET",
      methodFqn: "wix.cloud.autocms.tasks.v1.TaskService.ListTasks",
      packageName: PACKAGE_NAME,
      url: resolveWixCloudAutocmsTasksV1TaskServiceUrl({
        protoPath: "/v1",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "tasks.startedAt" },
            { path: "tasks.finishedAt" },
            {
              path: "tasks.updatePublishStatusOptions.scheduleDraftStatusOptions.date"
            },
            {
              path: "tasks.updatePublishStatusOptions.schedulePublishedStatusOptions.date"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listTasks;
}
function getTask(payload) {
  function __getTask({ host }) {
    const metadata = {
      entityFqdn: "wix.data.autocms.v2.background_task",
      method: "GET",
      methodFqn: "wix.cloud.autocms.tasks.v1.TaskService.GetTask",
      packageName: PACKAGE_NAME,
      url: resolveWixCloudAutocmsTasksV1TaskServiceUrl({
        protoPath: "/v1/{taskId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "task.startedAt" },
            { path: "task.finishedAt" },
            {
              path: "task.updatePublishStatusOptions.scheduleDraftStatusOptions.date"
            },
            {
              path: "task.updatePublishStatusOptions.schedulePublishedStatusOptions.date"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getTask;
}
function cancelTask(payload) {
  function __cancelTask({ host }) {
    const metadata = {
      entityFqdn: "wix.data.autocms.v2.background_task",
      method: "DELETE",
      methodFqn: "wix.cloud.autocms.tasks.v1.TaskService.CancelTask",
      packageName: PACKAGE_NAME,
      url: resolveWixCloudAutocmsTasksV1TaskServiceUrl({
        protoPath: "/v1/{taskId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "task.startedAt" },
            { path: "task.finishedAt" },
            {
              path: "task.updatePublishStatusOptions.scheduleDraftStatusOptions.date"
            },
            {
              path: "task.updatePublishStatusOptions.schedulePublishedStatusOptions.date"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __cancelTask;
}

// src/data-autocms-v2-background-task-tasks.universal.ts
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNKNOWN"] = "UNKNOWN";
  Type2["DELETE_BY_QUERY"] = "DELETE_BY_QUERY";
  Type2["COPY_FIELD_DATA"] = "COPY_FIELD_DATA";
  Type2["EDIT_FIELD_DATA"] = "EDIT_FIELD_DATA";
  Type2["UPDATE_PUBLISH_STATUS"] = "UPDATE_PUBLISH_STATUS";
  return Type2;
})(Type || {});
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["NEW"] = "NEW";
  Status2["RUNNING"] = "RUNNING";
  Status2["COMPLETED"] = "COMPLETED";
  Status2["FAILED"] = "FAILED";
  return Status2;
})(Status || {});
var Environment = /* @__PURE__ */ ((Environment2) => {
  Environment2["UNKNOWN_ENVIRONMENT"] = "UNKNOWN_ENVIRONMENT";
  Environment2["SANDBOX"] = "SANDBOX";
  Environment2["SANDBOX_PREFERRED"] = "SANDBOX_PREFERRED";
  Environment2["LIVE"] = "LIVE";
  return Environment2;
})(Environment || {});
var Operation = /* @__PURE__ */ ((Operation2) => {
  Operation2["UNDEFINED"] = "UNDEFINED";
  Operation2["SET_PUBLISHED_STATUS"] = "SET_PUBLISHED_STATUS";
  Operation2["SET_DRAFT_STATUS"] = "SET_DRAFT_STATUS";
  Operation2["SCHEDULE_PUBLISHED_STATUS"] = "SCHEDULE_PUBLISHED_STATUS";
  Operation2["SCHEDULE_DRAFT_STATUS"] = "SCHEDULE_DRAFT_STATUS";
  Operation2["CANCEL_SCHEDULING"] = "CANCEL_SCHEDULING";
  return Operation2;
})(Operation || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createTask2(task) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ task });
  const reqOpts = createTask(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.task;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { task: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["task"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listTasks2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    paging: options?.paging
  });
  const reqOpts = listTasks(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { paging: "$[0].paging" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getTask2(taskId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ taskId });
  const reqOpts = getTask(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data)?.task;
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { taskId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["taskId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function cancelTask2(taskId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({ taskId });
  const reqOpts = cancelTask(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { taskId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["taskId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/data-autocms-v2-background-task-tasks.public.ts
function createTask3(httpClient) {
  return (task) => createTask2(
    task,
    // @ts-ignore
    { httpClient }
  );
}
function listTasks3(httpClient) {
  return (options) => listTasks2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function getTask3(httpClient) {
  return (taskId) => getTask2(
    taskId,
    // @ts-ignore
    { httpClient }
  );
}
function cancelTask3(httpClient) {
  return (taskId) => cancelTask2(
    taskId,
    // @ts-ignore
    { httpClient }
  );
}

// src/data-autocms-v2-background-task-tasks.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
var createTask4 = /* @__PURE__ */ createRESTModule(createTask3);
var listTasks4 = /* @__PURE__ */ createRESTModule(listTasks3);
var getTask4 = /* @__PURE__ */ createRESTModule(getTask3);
var cancelTask4 = /* @__PURE__ */ createRESTModule(cancelTask3);
export {
  Environment,
  Operation,
  Status,
  Type,
  WebhookIdentityType,
  cancelTask4 as cancelTask,
  createTask4 as createTask,
  getTask4 as getTask,
  listTasks4 as listTasks
};
