// @wix/site-window/dist/esm/window.context.js
import { contextualizeHostModule } from "@wix/sdk-runtime/context";
import { windowRuntime as windowModule } from './window.runtime';
export const windowContext = {
    browserLocale: contextualizeHostModule(windowModule, 'browserLocale'),
    formFactor: contextualizeHostModule(windowModule, 'formFactor'),
    referrer: contextualizeHostModule(windowModule, 'referrer'),
    viewMode: contextualizeHostModule(windowModule, 'viewMode'),
    copyToClipboard: contextualizeHostModule(windowModule, 'copyToClipboard'),
    getAppPageData: contextualizeHostModule(windowModule, 'getAppPageData'),
    getBoundingRect: contextualizeHostModule(windowModule, 'getBoundingRect'),
    getCurrentGeolocation: contextualizeHostModule(windowModule, 'getCurrentGeolocation'),
    getRouterData: contextualizeHostModule(windowModule, 'getRouterData'),
    openAppLightbox: contextualizeHostModule(windowModule, 'openAppLightbox'),
    openLightbox: contextualizeHostModule(windowModule, 'openLightbox'),
    openModal: contextualizeHostModule(windowModule, 'openModal'),
    postMessage: contextualizeHostModule(windowModule, 'postMessage'),
    registerEventListener: contextualizeHostModule(windowModule, 'registerEventListener'),
    scrollBy: contextualizeHostModule(windowModule, 'scrollBy'),
    scrollTo: contextualizeHostModule(windowModule, 'scrollTo'),
    trackEvent: contextualizeHostModule(windowModule, 'trackEvent'),
};
