// @wix/wix-data-items-sdk/build/cjs/src/data-v2-data-item-items.context.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Environment = exports.onDataItemDeleted = exports.onDataItemUpdated = exports.onDataItemCreated = exports.withEnvironment = exports.filter = exports.queryReferenced = exports.aggregate = exports.bulkPatch = exports.patch = exports.search = exports.query = exports.replaceReferences = exports.removeReference = exports.insertReference = exports.isReferenced = exports.bulkRemove = exports.bulkSave = exports.bulkUpdate = exports.bulkInsert = exports.truncate = exports.remove = exports.get = exports.save = exports.update = exports.insert = void 0;
const data_v2_data_item_items_public_1 = require("./data-v2-data-item-items.public");
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
const event_definition_modules_1 = require("@wix/sdk-runtime/event-definition-modules");
exports.insert = (0, rest_modules_1.createRESTModule)(data_v2_data_item_items_public_1.insert);
exports.update = (0, rest_modules_1.createRESTModule)(data_v2_data_item_items_public_1.update);
exports.save = (0, rest_modules_1.createRESTModule)(data_v2_data_item_items_public_1.save);
exports.get = (0, rest_modules_1.createRESTModule)(data_v2_data_item_items_public_1.get);
exports.remove = (0, rest_modules_1.createRESTModule)(data_v2_data_item_items_public_1.remove);
exports.truncate = (0, rest_modules_1.createRESTModule)(data_v2_data_item_items_public_1.truncate);
exports.bulkInsert = (0, rest_modules_1.createRESTModule)(data_v2_data_item_items_public_1.bulkInsert);
exports.bulkUpdate = (0, rest_modules_1.createRESTModule)(data_v2_data_item_items_public_1.bulkUpdate);
exports.bulkSave = (0, rest_modules_1.createRESTModule)(data_v2_data_item_items_public_1.bulkSave);
exports.bulkRemove = (0, rest_modules_1.createRESTModule)(data_v2_data_item_items_public_1.bulkRemove);
exports.isReferenced = (0, rest_modules_1.createRESTModule)(data_v2_data_item_items_public_1.isReferenced);
exports.insertReference = (0, rest_modules_1.createRESTModule)(data_v2_data_item_items_public_1.insertReference);
exports.removeReference = (0, rest_modules_1.createRESTModule)(data_v2_data_item_items_public_1.removeReference);
exports.replaceReferences = (0, rest_modules_1.createRESTModule)(data_v2_data_item_items_public_1.replaceReferences);
exports.query = (0, rest_modules_1.createRESTModule)(data_v2_data_item_items_public_1.query);
exports.search = (0, rest_modules_1.createRESTModule)(data_v2_data_item_items_public_1.search);
exports.patch = (0, rest_modules_1.createRESTModule)(data_v2_data_item_items_public_1.patch);
exports.bulkPatch = (0, rest_modules_1.createRESTModule)(data_v2_data_item_items_public_1.bulkPatch);
exports.aggregate = (0, rest_modules_1.createRESTModule)(data_v2_data_item_items_public_1.aggregate);
exports.queryReferenced = (0, rest_modules_1.createRESTModule)(data_v2_data_item_items_public_1.queryReferenced);
exports.filter = (0, rest_modules_1.createRESTModule)(data_v2_data_item_items_public_1.filter);
exports.withEnvironment = (0, rest_modules_1.createRESTModule)(data_v2_data_item_items_public_1.withEnvironment);
/**
 * Triggered when a data item is inserted.
 */
exports.onDataItemCreated = (0, event_definition_modules_1.createEventModule)(data_v2_data_item_items_public_1.onDataItemCreated);
/**
 * Triggered when a data item is updated or patched.
 */
exports.onDataItemUpdated = (0, event_definition_modules_1.createEventModule)(data_v2_data_item_items_public_1.onDataItemUpdated);
/**
 * Triggered when a data item is deleted or truncated.
 */
exports.onDataItemDeleted = (0, event_definition_modules_1.createEventModule)(data_v2_data_item_items_public_1.onDataItemDeleted);
var data_v2_data_item_items_universal_1 = require("./data-v2-data-item-items.universal");
Object.defineProperty(exports, "Environment", { enumerable: true, get: function () { return data_v2_data_item_items_universal_1.Environment; } });
__exportStar(require("./data-v2-data-item-items.types"), exports);
