// @wix/wix-data-items-sdk/build/cjs/src/data-v2-data-item-items.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.insert = insert;
exports.update = update;
exports.save = save;
exports.get = get;
exports.remove = remove;
exports.truncate = truncate;
exports.bulkInsert = bulkInsert;
exports.patch = patch;
exports.bulkPatch = bulkPatch;
exports.bulkUpdate = bulkUpdate;
exports.bulkSave = bulkSave;
exports.bulkRemove = bulkRemove;
exports.isReferenced = isReferenced;
exports.insertReference = insertReference;
exports.removeReference = removeReference;
exports.replaceReferences = replaceReferences;
exports.search = search;
/* eslint-disable @typescript-eslint/no-empty-object-type */
const api = __importStar(require("./data-items-api-client"));
function insert() {
    return buildPublicMethodMetaInfo(api.insertDataItemRequestOptions);
}
function update() {
    return buildPublicMethodMetaInfo(api.updateDataItemRequestOptions);
}
function save() {
    return buildPublicMethodMetaInfo(api.saveDataItemRequestOptions);
}
function get() {
    return buildPublicMethodMetaInfo(api.getDataItemRequestOptions, { dataItemId: ':dataItemId' }, { dataItemId: 'dataItemId' });
}
function remove() {
    return buildPublicMethodMetaInfo(api.removeDataItemRequestOptions);
}
function truncate() {
    return buildPublicMethodMetaInfo(api.truncateDataItemsRequestOptions);
}
function bulkInsert() {
    return buildPublicMethodMetaInfo(api.bulkInsertDataItemsRequestOptions);
}
function patch() {
    return buildPublicMethodMetaInfo(api.patchDataItemRequestOptions);
}
function bulkPatch() {
    return buildPublicMethodMetaInfo(api.bulkPatchDataItemsRequestOptions);
}
function bulkUpdate() {
    return buildPublicMethodMetaInfo(api.bulkUpdateDataItemsRequestOptions);
}
function bulkSave() {
    return buildPublicMethodMetaInfo(api.bulkSaveDataItemsRequestOptions);
}
function bulkRemove() {
    return buildPublicMethodMetaInfo(api.bulkRemoveDataItemsRequestOptions);
}
function isReferenced() {
    return buildPublicMethodMetaInfo(api.isReferencedDataItemRequestOptions);
}
function insertReference() {
    return buildPublicMethodMetaInfo(api.bulkInsertDataItemsRequestOptions);
}
function removeReference() {
    return buildPublicMethodMetaInfo(api.bulkRemoveDataItemReferencesRequestOptions);
}
function replaceReferences() {
    return buildPublicMethodMetaInfo(api.replaceDataItemReferencesRequestOptions);
}
function search() {
    return buildPublicMethodMetaInfo(api.searchDataItemsRequestOptions);
}
function buildPublicMethodMetaInfo(requestOptionsParams, payload = {}, pathParams = {}) {
    return {
        getUrl: (context) => api.reqOptions(requestOptionsParams, payload)(context).url,
        httpMethod: requestOptionsParams.method,
        path: requestOptionsParams.httpPath,
        pathParams,
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
