// @wix/wix-data-items-sdk/build/cjs/src/data-items-api-client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.reqOptions = exports.searchDataItemsRequestOptions = exports.queryReferencedDataItemsRequestOptions = exports.aggregateDataItemsRequestOptions = exports.queryDataItemsRequestOptions = exports.queryDistinctValuesRequestOptions = exports.countDataItemsRequestOptions = exports.replaceDataItemReferencesRequestOptions = exports.bulkRemoveDataItemReferencesRequestOptions = exports.bulkInsertDataItemReferencesRequestOptions = exports.isReferencedDataItemRequestOptions = exports.bulkRemoveDataItemsRequestOptions = exports.bulkSaveDataItemsRequestOptions = exports.bulkPatchDataItemsRequestOptions = exports.bulkUpdateDataItemsRequestOptions = exports.bulkInsertDataItemsRequestOptions = exports.truncateDataItemsRequestOptions = exports.removeDataItemRequestOptions = exports.saveDataItemRequestOptions = exports.patchDataItemRequestOptions = exports.updateDataItemRequestOptions = exports.insertDataItemRequestOptions = exports.getDataItemRequestOptions = exports.WixDataItemsHttpClient = void 0;
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
class WixDataItemsHttpClient {
    constructor(httpClient, interceptor, sideEffects) {
        this.httpClient = httpClient;
        this.interceptor = interceptor;
        this.sideEffects = sideEffects;
    }
    getDataItem(request) {
        return this.execHttpCall(exports.getDataItemRequestOptions, request);
    }
    insertDataItem(request) {
        return this.execHttpCall(exports.insertDataItemRequestOptions, request);
    }
    updateDataItem(request) {
        return this.execHttpCall(exports.updateDataItemRequestOptions, request);
    }
    patchDataItem(request) {
        return this.execHttpCall(exports.patchDataItemRequestOptions, request);
    }
    saveDataItem(request) {
        return this.execHttpCall(exports.saveDataItemRequestOptions, request);
    }
    removeDataItem(request) {
        return this.execHttpCall(exports.removeDataItemRequestOptions, request);
    }
    truncateDataItems(request) {
        return this.execHttpCall(exports.truncateDataItemsRequestOptions, request);
    }
    bulkInsertDataItems(request) {
        return this.execHttpCall(exports.bulkInsertDataItemsRequestOptions, request);
    }
    bulkPatchDataItems(request) {
        return this.execHttpCall(exports.bulkPatchDataItemsRequestOptions, request);
    }
    bulkUpdateDataItems(request) {
        return this.execHttpCall(exports.bulkUpdateDataItemsRequestOptions, request);
    }
    bulkSaveDataItems(request) {
        return this.execHttpCall(exports.bulkSaveDataItemsRequestOptions, request);
    }
    bulkRemoveDataItems(request) {
        return this.execHttpCall(exports.bulkRemoveDataItemsRequestOptions, request);
    }
    isReferencedDataItem(request) {
        return this.execHttpCall(exports.isReferencedDataItemRequestOptions, request);
    }
    bulkInsertDataItemReferences(request) {
        return this.execHttpCall(exports.bulkInsertDataItemReferencesRequestOptions, request);
    }
    bulkRemoveDataItemReferences(request) {
        return this.execHttpCall(exports.bulkRemoveDataItemReferencesRequestOptions, request);
    }
    replaceDataItemReferences(request) {
        return this.execHttpCall(exports.replaceDataItemReferencesRequestOptions, request);
    }
    countDataItems(request) {
        return this.execHttpCall(exports.countDataItemsRequestOptions, request);
    }
    queryDistinctValues(request) {
        return this.execHttpCall(exports.queryDistinctValuesRequestOptions, request);
    }
    queryDataItems(request) {
        return this.execHttpCall(exports.queryDataItemsRequestOptions, request);
    }
    aggregateDataItems(request) {
        return this.execHttpCall(exports.aggregateDataItemsRequestOptions, request);
    }
    queryReferencedDataItems(request) {
        return this.execHttpCall(exports.queryReferencedDataItemsRequestOptions, request);
    }
    searchDataItems(request) {
        return this.execHttpCall(exports.searchDataItemsRequestOptions, request);
    }
    async execHttpCall(params, payload) {
        const reqOpts = (0, exports.reqOptions)(params, payload);
        this.sideEffects?.onSiteCall?.();
        try {
            const result = await this.httpClient.request((ctx) => {
                const req = reqOpts(ctx);
                const intercepted = this.interceptor(req);
                return intercepted;
            });
            this.sideEffects?.onSuccess?.(result);
            return result;
        }
        catch (err) {
            this.sideEffects?.onError?.(err);
            throw err;
        }
    }
}
exports.WixDataItemsHttpClient = WixDataItemsHttpClient;
exports.getDataItemRequestOptions = {
    method: 'GET',
    methodName: 'GetDataItem',
    httpPath: '/v2/items/{dataItemId}',
};
exports.insertDataItemRequestOptions = {
    method: 'POST',
    methodName: 'InsertDataItem',
    httpPath: '/v2/items',
};
exports.updateDataItemRequestOptions = {
    method: 'POST',
    methodName: 'UpdateDataItem',
    httpPath: '/v2/items/update',
};
exports.patchDataItemRequestOptions = {
    method: 'POST',
    methodName: 'PatchDataItem',
    httpPath: '/v2/items/patch',
};
exports.saveDataItemRequestOptions = {
    method: 'POST',
    methodName: 'SaveDataItem',
    httpPath: '/v2/items/save',
};
exports.removeDataItemRequestOptions = {
    method: 'POST',
    methodName: 'RemoveDataItem',
    httpPath: '/v2/items/remove',
};
exports.truncateDataItemsRequestOptions = {
    method: 'POST',
    methodName: 'TruncateDataItems',
    httpPath: '/v2/items/truncate',
};
exports.bulkInsertDataItemsRequestOptions = {
    method: 'POST',
    methodName: 'BulkInsertDataItems',
    httpPath: '/v2/bulk/items/insert',
};
exports.bulkUpdateDataItemsRequestOptions = {
    method: 'POST',
    methodName: 'BulkUpdateDataItems',
    httpPath: '/v2/bulk/items/update',
};
exports.bulkPatchDataItemsRequestOptions = {
    method: 'POST',
    methodName: 'BulkPatchDataItems',
    httpPath: '/v2/bulk/items/patch',
};
exports.bulkSaveDataItemsRequestOptions = {
    method: 'POST',
    methodName: 'BulkSaveDataItems',
    httpPath: '/v2/bulk/items/save',
};
exports.bulkRemoveDataItemsRequestOptions = {
    method: 'POST',
    methodName: 'BulkRemoveDataItems',
    httpPath: '/v2/bulk/items/remove',
};
exports.isReferencedDataItemRequestOptions = {
    method: 'POST',
    methodName: 'IsReferencedDataItem',
    httpPath: '/v2/items/is-referenced',
};
exports.bulkInsertDataItemReferencesRequestOptions = {
    method: 'POST',
    methodName: 'BulkInsertDataItemReferences',
    httpPath: '/v2/bulk/items/insert-references',
};
exports.bulkRemoveDataItemReferencesRequestOptions = {
    method: 'POST',
    methodName: 'BulkRemoveDataItemReferences',
    httpPath: '/v2/bulk/items/remove-references',
};
exports.replaceDataItemReferencesRequestOptions = {
    method: 'POST',
    methodName: 'ReplaceDataItemReferences',
    httpPath: '/v2/items/replace-references',
};
exports.countDataItemsRequestOptions = {
    method: 'POST',
    methodName: 'CountDataItems',
    httpPath: '/v2/items/count',
};
exports.queryDistinctValuesRequestOptions = {
    method: 'POST',
    methodName: 'QueryDistinctValues',
    httpPath: '/v2/items/query-distinct-values',
};
exports.queryDataItemsRequestOptions = {
    method: 'POST',
    methodName: 'QueryDataItems',
    httpPath: '/v2/items/query',
};
exports.aggregateDataItemsRequestOptions = {
    method: 'POST',
    methodName: 'AggregateDataItems',
    httpPath: '/v2/items/aggregate',
};
exports.queryReferencedDataItemsRequestOptions = {
    method: 'POST',
    methodName: 'QueryReferencedDataItems',
    httpPath: '/v2/items/query-referenced',
};
exports.searchDataItemsRequestOptions = {
    method: 'POST',
    methodName: 'SearchDataItems',
    httpPath: '/v2/items/search',
};
const reqOptions = ({ method, methodName, httpPath }, payload) => {
    const useSearchParams = method === 'GET' || method === 'DELETE';
    return ({ host }) => ({
        entityFqdn: 'wix.data.v2.data_item',
        method,
        methodFqn: `com.wixpress.cloud.data.api.data.DataItemService.${methodName}`,
        url: resolveServiceUrl({
            protoPath: httpPath,
            data: payload,
            host,
        }),
        data: useSearchParams ? undefined : payload,
        params: useSearchParams ? (0, rest_modules_1.toURLSearchParams)(payload) : undefined,
    });
};
exports.reqOptions = reqOptions;
const resolveServiceUrl = (opts) => {
    const domainToMappings = {
        'api._api_base_domain_': [
            {
                srcPath: '/cloud-data',
                destPath: '',
            },
        ],
        'cloud-data.wix-code.com': [
            {
                srcPath: '',
                destPath: '',
            },
        ],
        _: [
            {
                srcPath: '/_api/cloud-data',
                destPath: '',
            },
        ],
        'www._base_domain_': [
            {
                srcPath: '/_api/cloud-data',
                destPath: '',
            },
            {
                srcPath: '/_api/data',
                destPath: '',
            },
        ],
        'dev._base_domain_': [
            {
                srcPath: '/_api/cloud-data',
                destPath: '',
            },
        ],
        'bo._base_domain_': [
            {
                srcPath: '/_api/cloud-data/v2',
                destPath: '/v2',
            },
        ],
        'wixbo.ai': [
            {
                srcPath: '/_api/cloud-data/v2',
                destPath: '/v2',
            },
        ],
        'manage._base_domain_': [
            {
                srcPath: '/_api/cloud-data',
                destPath: '',
            },
            {
                srcPath: '/_api/data',
                destPath: '',
            },
        ],
        'editor._base_domain_': [
            {
                srcPath: '/_api/cloud-data/v2',
                destPath: '/v2',
            },
        ],
        'blocks._base_domain_': [
            {
                srcPath: '/_api/cloud-data/v2',
                destPath: '/v2',
            },
        ],
        'create.editorx': [
            {
                srcPath: '/_api/cloud-data/v2/bulk',
                destPath: '/v2/bulk',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/wix-data/v2',
                destPath: '/v2',
            },
        ],
        '*.dev.wix-code.com': [
            {
                srcPath: '/_api/cloud-data/v2',
                destPath: '/v2',
            },
        ],
    };
    return (0, rest_modules_1.resolveUrl)(Object.assign(opts, { domainToMappings }));
};
