// @wix/wix-data-items-sdk/build/cjs/src/data-v2-data-item-items.public.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Environment = exports.onDataItemDeleted = exports.onDataItemUpdated = exports.onDataItemCreated = exports.__metadata = void 0;
exports.withEnvironment = withEnvironment;
exports.insert = insert;
exports.update = update;
exports.save = save;
exports.get = get;
exports.remove = remove;
exports.truncate = truncate;
exports.bulkInsert = bulkInsert;
exports.bulkUpdate = bulkUpdate;
exports.bulkSave = bulkSave;
exports.bulkRemove = bulkRemove;
exports.isReferenced = isReferenced;
exports.insertReference = insertReference;
exports.removeReference = removeReference;
exports.replaceReferences = replaceReferences;
exports.query = query;
exports.search = search;
exports.patch = patch;
exports.bulkPatch = bulkPatch;
exports.aggregate = aggregate;
exports.queryReferenced = queryReferenced;
exports.filter = filter;
const sdk_types_1 = require("@wix/sdk-types");
const data_v2_data_item_items_universal_1 = require("./data-v2-data-item-items.universal");
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const wix_data_items_common_1 = require("@wix/wix-data-items-common");
exports.__metadata = { PACKAGE_NAME: '@wix/data' };
function withEnvironment(httpClient) {
    return (environment) => (0, data_v2_data_item_items_universal_1.withEnvironment)(environment, 
    // @ts-expect-error
    { httpClient });
}
function insert(httpClient) {
    return (dataCollectionId, dataItem, options) => (0, data_v2_data_item_items_universal_1.insert)(dataCollectionId, dataItem, options, 
    // @ts-expect-error
    { httpClient });
}
function update(httpClient) {
    return (dataCollectionId, data, options) => (0, data_v2_data_item_items_universal_1.update)(dataCollectionId, data, options, 
    // @ts-expect-error
    { httpClient });
}
function save(httpClient) {
    return (dataCollectionId, data, options) => (0, data_v2_data_item_items_universal_1.save)(dataCollectionId, data, options, 
    // @ts-expect-error
    { httpClient });
}
function get(httpClient) {
    return (dataCollectionId, dataItemId, options) => (0, data_v2_data_item_items_universal_1.get)(dataCollectionId, dataItemId, options, 
    // @ts-expect-error
    { httpClient });
}
function remove(httpClient) {
    return (dataCollectionId, dataItemId, options) => (0, data_v2_data_item_items_universal_1.remove)(dataCollectionId, dataItemId, options, 
    // @ts-expect-error
    { httpClient });
}
function truncate(httpClient) {
    return (dataCollectionId) => (0, data_v2_data_item_items_universal_1.truncate)(dataCollectionId, 
    // @ts-expect-error
    { httpClient });
}
function bulkInsert(httpClient) {
    return (dataCollectionId, items, options) => (0, data_v2_data_item_items_universal_1.bulkInsert)(dataCollectionId, items, options, 
    // @ts-expect-error
    { httpClient });
}
function bulkUpdate(httpClient) {
    return (dataCollectionId, items, options) => (0, data_v2_data_item_items_universal_1.bulkUpdate)(dataCollectionId, items, options, 
    // @ts-expect-error
    { httpClient });
}
function bulkSave(httpClient) {
    return (dataCollectionId, items, options) => (0, data_v2_data_item_items_universal_1.bulkSave)(dataCollectionId, items, options, 
    // @ts-expect-error
    { httpClient });
}
function bulkRemove(httpClient) {
    return (dataCollectionId, itemIds, options) => (0, data_v2_data_item_items_universal_1.bulkRemove)(dataCollectionId, itemIds, options, 
    // @ts-expect-error
    { httpClient });
}
function isReferenced(httpClient) {
    return (dataCollectionId, propertyName, referringItem, referencedItem, options) => (0, data_v2_data_item_items_universal_1.isReferenced)(dataCollectionId, propertyName, referringItem, referencedItem, options, 
    // @ts-expect-error
    { httpClient });
}
function insertReference(httpClient) {
    return (dataCollectionId, propertyName, referringItem, referencedItem, options) => (0, data_v2_data_item_items_universal_1.insertReference)(dataCollectionId, propertyName, referringItem, referencedItem, options, 
    // @ts-expect-error
    { httpClient });
}
function removeReference(httpClient) {
    return (dataCollectionId, propertyName, referringItem, referencedItem, options) => (0, data_v2_data_item_items_universal_1.removeReference)(dataCollectionId, propertyName, referringItem, referencedItem, options, 
    // @ts-expect-error
    { httpClient });
}
function replaceReferences(httpClient) {
    return (dataCollectionId, propertyName, referringItem, referencedItem, options) => (0, data_v2_data_item_items_universal_1.replaceReferences)(dataCollectionId, propertyName, referringItem, referencedItem, options, 
    // @ts-expect-error
    { httpClient });
}
function query(httpClient) {
    return (dataCollectionId) => (0, data_v2_data_item_items_universal_1.query)(dataCollectionId, 
    // @ts-expect-error
    { httpClient });
}
function search(httpClient) {
    return (dataCollectionId) => (0, data_v2_data_item_items_universal_1.search)(dataCollectionId, 
    // @ts-expect-error
    { httpClient });
}
function patch(httpClient) {
    return (dataCollectionId, itemId) => (0, data_v2_data_item_items_universal_1.patch)(dataCollectionId, itemId, 
    // @ts-expect-error
    { httpClient });
}
function bulkPatch(httpClient) {
    return (dataCollectionId, itemIds) => (0, data_v2_data_item_items_universal_1.bulkPatch)(dataCollectionId, itemIds, 
    // @ts-expect-error
    { httpClient });
}
function aggregate(httpClient) {
    return (dataCollectionId) => (0, data_v2_data_item_items_universal_1.aggregate)(dataCollectionId, 
    // @ts-expect-error
    { httpClient });
}
function queryReferenced(httpClient) {
    return (dataCollectionId, holdingItem, relationshipAttribute, options) => (0, data_v2_data_item_items_universal_1.queryReferenced)(dataCollectionId, holdingItem, relationshipAttribute, options, 
    // @ts-expect-error
    { httpClient });
}
function filter(httpClient) {
    return () => (0, data_v2_data_item_items_universal_1.filter)(
    // @ts-expect-error
    { httpClient });
}
exports.onDataItemCreated = (0, sdk_types_1.EventDefinition)('wix.data.v2.data_item_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)((0, wix_data_items_common_1.decode)(event), [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [{ path: 'metadata.eventTime' }],
    },
]), ['entity.data']))();
exports.onDataItemUpdated = (0, sdk_types_1.EventDefinition)('wix.data.v2.data_item_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)((0, wix_data_items_common_1.decode)(event), [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [{ path: 'metadata.eventTime' }],
    },
]), ['entity.data']))();
exports.onDataItemDeleted = (0, sdk_types_1.EventDefinition)('wix.data.v2.data_item_deleted', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)((0, wix_data_items_common_1.decode)(event), [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [{ path: 'metadata.eventTime' }],
    },
]), ['entity.data']))();
var data_v2_data_item_items_universal_2 = require("./data-v2-data-item-items.universal");
Object.defineProperty(exports, "Environment", { enumerable: true, get: function () { return data_v2_data_item_items_universal_2.Environment; } });
__exportStar(require("./data-v2-data-item-items.types"), exports);
